src/with_clause_parser.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-APACHE for a copy of the license.
+ */
+#include <postgres.h>
+#include <c.h>
+
+#include <nodes/parsenodes.h>
+
+/* find the length of a statically sized array */
+#define TS_ARRAY_LEN(array) (sizeof(array) / sizeof(*array))
+
```

Comment:
So I think there is a bit of inconsistency here: `WithClauseArg` is usually used as an array of ` WithClauseArg`s but for `WithClauseParse` there is are 2 arrays /inside/ `WithClauseParse`. Is there a good reason for this? Why not return an array of `WithClauseParse` instead?

Reply:
mostly so I could `memcpy(fcinfo.args, WithClauseResult.results, ...)`, and waste less space on padding. So _a_ reason, though I'm not sure how compelling of one. (FWIW, I see the choice of returning an array of structs versus a struct of arrays to be fairly arbitrary one, and I wouldn't expect there to be any consistency between the input and output layout).